### PR TITLE
Redo

### PR DESCRIPTION
**Pull request by anmolint**

### CODE REVIEW

File: controller/login.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    try{
+        console.log(req.body);
+        let user =await data.findOne({username: req.body.username});
+        console.log(user)
+        if(user){
+        let decryption = await crypt.compare(req.body.password,user.password);
+        console.log(decryption);
+        if(decryption == true )
+        {
+            let token = jwt.sign({ user_id: user._id},'abcd')
```
Comment (AbhiSeCode) [inline controller/login.js]: Store Secret key of JWT in .env file and use it from there.

File: controller/login.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const { default: async } = require("async");
+const data = require('../model/user')
+const crypt = require("bcrypt");
+const jwt = require('jsonwebtoken');
+const logIn = async(req,res) =>{
+    try{
+        console.log(req.body);
```
Comment (AbhiSeCode) [inline controller/login.js]: Remove consoles from code.

File: controller/network.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require("mongoose");
+main()
+  .then(() => {})
+  .catch((err) => console.log(err));
+async function main() {
+  a = await mongoose.connect(
+    "mongodb+srv://Anmol:baloni@cluster0.v1fvg.mongodb.net/myFirstDatabase?retryWrites=true&w=majority"
```
Comment (AbhiSeCode) [inline controller/network.js]: Store SRV too in .env

File: controller/returnData.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    try{
+        console.log(req.query)
+        const { page = req.query.id, limit = 10 } = req.query;
+        const posts = await data.find()
+      .limit(limit * 1)
+      .skip((page - 1) * limit)
+      .exec();
+        
+      const count = await data.countDocuments();
+      res.send(count,posts)
```
Comment (AbhiSeCode) [inline controller/returnData.js]: Why are we sending two different response. ?

File: controller/usersRegister.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  try {
+    console.log(req.body);
+    if (req.body.password != req.body.confpass) {
+      console.error;
+    }
+    let saltRounds = await crypt.genSalt(10);
+    console.log(saltRounds);
+    let encrypt = await crypt.hashSync(req.body.password, saltRounds);
+    console.log(encrypt);
+    let w = await data.create({
```
Comment (AbhiSeCode) [inline controller/usersRegister.js]: Name variable that make more sense.

File: controller/usersRegister.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      username: req.body.username,
+      password: encrypt,
+      email: req.body.email,
+      firstname: req.body.firstname,
+      lastname: req.body.lastname,
+    });
+    console.log("sucessfully created data");
+    res.send({ token: w });
```
Comment (AbhiSeCode) [inline controller/usersRegister.js]: We should send JWT token and not user data when user is signed up.

File: model/user.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    username: { type: "string", unique: true },
+    password: "string",
+    confpass: "string",
+    email: { type: "string", unique: true },
+    firstname: "string",
+    lastname: "string",
+  },
+]);
+const datavalue = mongoose.model("dbData", data_value);
+module.exports = datavalue;
```
Comment (AbhiSeCode) [inline model/user.js:14]: Same naming issue, collection name should make sense.

File: controller/getUser.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+const { default: async } = require("async");
+const data = require('../model/user')
+const jwt = require('jsonwebtoken');
+const giveData = async(req,res) =>{
+    try{
+        console.log(req.query)
+        let decoded = jwt.verify(req.query.id, 'abcd');
```
Comment (AbhiSeCode) [inline controller/getUser.js]: We are not going to fetch token from query string, we will fetch token from header and then will verify it. Also create a middleware to verify JWT.

File: controller/deleteUser.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
+const { default: async } = require("async");
```
Comment (AbhiSeCode) [inline controller/deleteUser.js]: No need to make a new file for every controller. You just need to create a userController file and write all the controllers related to user routes there.

File: .env [hunk 10]
```diff
@@ -0,0 +1,3 @@
+key =abcd
+mongoUri = mongodb+srv://Anmol:baloni@cluster0.v1fvg.mongodb.net/myFirstDatabase?retryWrites=true&w=majority
+keymd = 12345
```
Comment (abhay-sky) [inline .env]: do not push env in the code

File: controller/userController.js [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+    let token = registration._id
+    let genrate =await userTokengenrator.create({
+      user_id:token,
+      // access_token:md5(process.env.keymd)
+      access_token:md5(new Date())
+    })
+    //jwt.sign({ registration: registration._id }, process.env.key, {
+    //   expiresIn: "1h",
+    // });
+    console.log(genrate);
```
Comment (abhay-sky) [inline controller/userController.js]: remove un-necessary consoles

File: controller/userController.js [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+        let userid = await data.findById(user._)
+        let token = await userTokengenrator.create({
+          user_id:userid,
+          access_token:md5(new Date())
+
+        });
+        // let token = jwt.sign({ user_id: user._id }, process.env.key, {
+        //   expiresIn: "1h",
+        // });
+        res.send(md5(new Date()));
```
Comment (abhay-sky) [inline controller/userController.js]: need to check api's with this token is it even working ?

File: middleware/authUser.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+// const jwt = require("jsonwebtoken");
+const tokenVerification = async (req, res, next) => {
+  try {
+    const tkn = req.headers.authorization;
+    if (!tkn) {
+      res.send(" auth token required");
+    } else {
+      let tokenAutentication = tokenAutenticator.find(
+        req.headers.authorization
+      );
```
Comment (abhay-sky) [inline middleware/authUser.js]: have you test this middleware

**Pull request merged**